python/src/block/ProbabilityTableBlock.python.cpp

```diff
@@ -0,0 +1,152 @@
+// system includes
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+// local includes
+#include "ACEtk/block/ProbabilityTableBlock.hpp"
+#include "views.hpp"
+#include "definitions.hpp"
+
+// namespace aliases
+namespace python = pybind11;
+
+namespace block {
+
+void wrapProbabilityTableBlock( python::module& module, python::module& ) {
+
+  // type aliases
+  using Block = njoy::ACEtk::block::ProbabilityTableBlock;
+  using ProbabilityTable = njoy::ACEtk::block::ProbabilityTable;
+
+  // wrap views created by this block
+
+  // create the block
+  python::class_< Block > block(
+
+    module,
+    "ProbabilityTableBlock",
+    "he continuous energy UNR block with the unresolved resonance probability tables"
+  );
+
+  // wrap the block
+  block
+  .def(
+
+    python::init< unsigned int, int, int, unsigned int,
+                  std::vector< ProbabilityTable > >(),
+    python::arg( "interpolation" ),
+    python::arg( "inelastic" ),
+    python::arg( "absorption" ),
+    python::arg( "type" ),
+    python::arg( "tables" ),
+    "Initialise the block\n\n"
+    "Arguments:\n"
+    "    self             the block\n"
+    "    interpolation    the interpolation type\n"
+    "    inelastic        the inelastic competition flag\n"
+    "    absorption       the absorption competition flag\n"
+    "    type             the probability table type\n"
+    "    tables           the probability tables"
+  )
+  .def_property_readonly(
+
+    "N",
+    &Block::N,
+    "The number of incident energy values"
+  )
+  .def_property_readonly(
+
+    "number_incident_energies",
+    &Block::numberIncidentEnergies,
+    "The number of incident energy values"
+  )
+  .def_property_readonly(
+
+    "M",
+    &Block::M,
+    "The number of bins in each table"
+  )
+  .def_property_readonly(
+
+    "number_bins",
+    &Block::numberBins,
+    "The number of bins in each table"
+  )
+  .def_property_readonly(
+
+    "INT",
+    &Block::INT,
+    "The interpolation flag"
+  )
+  .def_property_readonly(
+
+    "interpolation",
+    &Block::interpolation,
+    "The interpolation flag"
+  )
+  .def_property_readonly(
+
+    "ILF",
+    &Block::ILF,
+    "The inelastic competition flag"
+  )
+  .def_property_readonly(
+
+    "inelastic_competion",
+    &Block::inelasticCompetion,
+    "The inelastic competition flag"
+  )
+  .def_property_readonly(
+
+    "IOA",
+    &Block::IOA,
+    "The absorption competition flag"
+  )
+  .def_property_readonly(
+
+    "absorption_competion",
+    &Block::absorptionCompetion,
```

Comment:
This variable and associated documentation seems to be missing a `t`.

Reply:
How is this possible? It's not even an English word. It also happened for the inelastic competition. I updated the C++ and Python interface and updated tests for both.